src/components/Column/ColumnSettings.tsx

```diff
@@ -36,109 +41,89 @@ export const ColumnSettings: FC<ColumnSettingsProps> = ({id, name, color, visibl
     dispatch(Actions.createColumnOptimistically({id: TEMPORARY_COLUMN_ID, name: "", color: randomColor, visible: false, index: columnIndex}));
   };
 
+  useEffect(() => {
+    const handleKeyPress = (e: KeyboardEvent) => {
+      if (e.key === "Escape" && !openedColorPicker) {
+        props.closeColumnSettings();
+      }
+    };
+
+    document.addEventListener("keydown", handleKeyPress, false); // trigger in bubble phase
+    return () => document.removeEventListener("keydown", handleKeyPress, false);
+  }, [openedColorPicker, props]);
+
+  const renderColorPicker = () =>
+    openedColorPicker ? (
+      <ColorPicker
+        id={props.id}
+        name={props.name}
+        visible={props.visible}
+        index={props.index}
+        color={props.color}
+        onClose={props.onClose}
+        colors={COLOR_ORDER}
+        closeColorPicker={() => setOpenedColorPicker(false)}
+      />
+    ) : (
+      <span className="column__header-color-option column__header-color-option-selected" />
+    );
+
+  const menuItems = [
```

Comment:
```suggestion
  const menuItems: MiniMenuItem[] = [
```

of course the type needs to exported from `MiniMenu` for this to work, but I prefer the compiler knowing the correct type